ttnn/cpp/ttnn/operations/pool/upsample/device/upsample_program_factory_multicore.cpp

```diff
@@ -13,13 +15,106 @@
+                }
+            }
+        }
+    }
+
+    std::vector<uint16_t> logical_core_to_stick_map;
+    size_t logical_core_to_stick_map_entry_size = 3;
+    size_t row_size = logical_core_to_stick_map_entry_size * in_w;
+    // Create map of core and respective offsets in input
+    for (uint32_t b = 0; b < batch_size; ++b) {
+        for (uint32_t h = 0; h < in_h; ++h) {
+            for (uint32_t w = 0; w < in_w; ++w, ++curr_stick) {
+                if (curr_stick == input_nsticks_per_core) {
+                    curr_stick = 0, ++in_core;
+                }
+                if (is_height_sharded && is_col_major) {
+                    logical_core_to_stick_map.push_back(core_range[in_core][0]);
+                    logical_core_to_stick_map.push_back(core_range[in_core][1]);
+                } else {
+                    logical_core_to_stick_map.push_back(in_core);
+                    logical_core_to_stick_map.push_back(0);
+                }
+                logical_core_to_stick_map.push_back(curr_stick);
+            }
+            for (uint32_t j = 1; j < scale_factor_h; ++j) {
+                logical_core_to_stick_map.insert(
+                    logical_core_to_stick_map.end(),
+                    logical_core_to_stick_map.end() - row_size,
+                    logical_core_to_stick_map.end());
+            }
+        }
+    }
+
+    std::vector<uint16_t> config_vector;
+
+    // Based on core calculate physical location of cores
+    CoreCoord core_coords;
+    if (is_height_sharded) {
+        for (size_t j = 0; j < logical_core_to_stick_map.size(); j += logical_core_to_stick_map_entry_size) {
+            CoreCoord core_coords;
+            if (is_col_major) {
+                core_coords = device->worker_core_from_logical_core(
+                    CoreCoord(logical_core_to_stick_map[j], logical_core_to_stick_map[j + 1]));
+            } else {
+                core_coords = device->worker_core_from_logical_core(
+                    CoreCoord(logical_core_to_stick_map[j] % ncores_x, logical_core_to_stick_map[j] / ncores_x));
+            }
+            config_vector.push_back(core_coords.x);
+            config_vector.push_back(core_coords.y);
+            config_vector.push_back(logical_core_to_stick_map[j + 2]);
+            config_vector.push_back(0);
+        }
+    } else {
+        for (size_t i = 0; i < ncores_x; i++) {
+            for (size_t j = 0; j < logical_core_to_stick_map.size(); j += logical_core_to_stick_map_entry_size) {
+                core_coords = device->worker_core_from_logical_core(CoreCoord(i, logical_core_to_stick_map[j]));
+                config_vector.push_back(core_coords.x);
+                config_vector.push_back(core_coords.y);
+                config_vector.push_back(logical_core_to_stick_map[j + 2]);
+                config_vector.push_back(0);
+            }
+        }
+    }
+    uint32_t elems_per_core = 4 * scale_factor_h * input_nsticks_per_core;
```

Comment:
Please have a constexpr with a descriptive name for 4.

Reply:
Done.